sys/parsoid.js

```diff
@@ -380,6 +380,22 @@ PSP.getRevisionInfo = function(restbase, req) {
     });
 };
 
+/**
+ * Internal check to see if it's okay to re-render a particular title in
+ * response to a no-cache request.
+ * @param {Request}
+ * @return {boolean} Whether re-rendering this title is okay.
+ */
+PSP._okayToRerender = function(req) {
+    var blackList = this.options.rerenderBlacklist;
+    if (blackList) {
+        return !blackList[req.params.domain]
+            || !blackList[req.params.domain][req.params.title];
```

Comment:
Malicious clients can also just request every revision of every article, focusing on the most expensive pages. This isn't meant as an effective defense against external requests. It's focused on requests from the job queue.